### PR TITLE
knowledge-graph to handle GraphQL queries

### LINKED ISSUE #19: PoC with graphql
~~Try out https://www.hypergraphql.org/ with the Jena endpoint~~
Implement a graphql endpoint for graph queries

### PR DESCRIPTION
**Pull request by jachro**
This PR introduces a new graph-service exposing a new GraphQL endpoint. At the moment the endpoint allows querying project lineage only but is easily extendable to provide more information in the future.

All the necessary details about the url, request and response format could be found in the service's README file.

There are following PR related to this one on the other repositories:
* https://github.com/SwissDataScienceCenter/renku/pull/587
* https://github.com/SwissDataScienceCenter/sdsc-dev-docs/pull/32

---

closes #19

### CODE REVIEW

File: helm-chart/renku-graph/values.yaml [hunk 1]
```diff
@@ -21,8 +21,8 @@ tokenRepository:
 
 triplesGenerator:
   image:
-    repository: renku/triples-generator
-    tag: 'latest'
+    repository: jachro/triples-generator
```
Comment (rokroskar) [inline helm-chart/renku-graph/values.yaml]: don't think you want to commit these

File: helm-chart/renku-graph/Chart.yaml [hunk 2]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 appVersion: '1.0'
 description: A Helm chart for renku graph services
 name: renku-graph
-version: 0.1.0
+version: 0.1.0-1c62490
```
Comment (rokroskar) [inline helm-chart/renku-graph/Chart.yaml]: probably bump this to 0.2.0

Comment (jachro) [reply helm-chart/renku-graph/Chart.yaml]: Yes, I was thinking about releasing all the services but maybe once they get merged to master.

File: helm-chart/renku-graph/values.yaml (hunk 1 above)
Comment (jachro) [reply helm-chart/renku-graph/values.yaml]: arghhh, again :)

File: helm-chart/renku-graph/Chart.yaml (hunk 2 above)
Comment (rokroskar) [reply helm-chart/renku-graph/Chart.yaml]: no, I mean the working version of the chart should be `>0.1.0`, i.e. `0.2.0-1234` is a development version of `0.2.0`.

Comment (jachro) [reply helm-chart/renku-graph/Chart.yaml]: Yes, that's true -> updated.

**Comment by jachro** [comment]
Hi @lorenzo-cavazzi 
I've just pushed a fix for the queries having variables. I was able to keep the scalar types as they were before but as a user, you can simply provide a primitive value. In other words, this kind of query works nicely now:
```
{ 
  "query": "query($projectPath: ProjectPath!) { 
    lineage(projectPath: $projectPath) {
      nodes { id label } 
      edges { source target } 
    } 
  }",
  "variables": {
    "projectPath": "rokroskar/vom_natt"
  }
}
```

**Comment by lorenzo-cavazzi** [comment]
@jachro now it's working :tada: 
Tomorrow I'll do a few more tests before approving the PR.

**Comment by lorenzo-cavazzi** [comment]
@jachro now it's working :tada: 
Tomorrow I'll do a few more tests before approving the PR.

**Pull request merged**